wrappers/clang-target-wrapper.sh

```diff
@@ -52,6 +53,24 @@ aarch64)
     # Dwarf is the default here.
     ;;
 esac
+case $TARGET_OS in
+mingw32uwp)
+    # the UWP target is for Windows 10
+    FLAGS="$FLAGS -D_WIN32_WINNT=0x0A00 -DWINVER=0x0A00"
+    # the UWP target can only use Windows Store APIs
+    FLAGS="$FLAGS -DWINAPI_FAMILY=WINAPI_FAMILY_APP"
+    # the Windows Store API only supports Windows Unicode (some rare ANSI ones are available)
+    FLAGS="$FLAGS -D_UNICODE -DUNICODE"
```

Comment:
Defining UNICODE without building with `-municode` could, in theory, mess up programs that use a `_tmain` entry point (i.e. the entry point ends up as `_wmain` but the right entry point is not passed to the linker). I think it could be acceptable, but adding `-municode` instead should be safer. (IIRC `-municode` only defines `UNICODE`, not `_UNICODE` though, so `-municode -D_UNICODE` could also be ok, in case both actually are needed.)

Reply:
It seems more correct, but it fails to link properly with this change
```
i686-w64-mingw32uwp-clang hello.c -o i686-mingw32uwp/hello.exe
lld-link: error: undefined symbol: _wWinMain@16
>>> referenced by libmingw32.a(lib32_libmingw32_a-crt0_w.o):(_wmain)
```

Meaning I probably need to compile a separate CRT for UWP/WinRT just for this call. Are there any other side effects ? Because we may not need this for now. I don't think we can generate UWP EXE just yet as they need to be packaged with Visual Studio.

Reply:
Ah right, if `-municode` is forced, the entry point to the executable _must_ be `_wmain` or `_tmain`. So it won't work with the regular test files.

The case of `_tmain` breaking if `-DUNICODE` is specified without `-municode` is much less hassle and mostly a theoretical cornercase I came up with. So let's skip that for now.

Reply:
I guess the remaining question then is, do you need both `UNICODE` and `_UNICODE`, or is the former enough? IIRC the mingw headers only look for `UNICODE`, is there anything that uses the latter?

Reply:
Looking at the VLC contribs, only `live555` relies on `_UNICODE` instead of `UNICODE`. It could be patched.

But the 2 defines are not actually the same https://stackoverflow.com/questions/7953025/why-both-unicode-and-unicode `UNICODE` triggers the Unicode API calls (we want), `_UNICODE` is more for tchar.h which is good to have but should be app specific (ie defined in the app that's building the code). So I will remove `_UNICODE`.

Reply:
Ah, right. In that case I'd be ok with keeping both as well, but it's up to you.

Reply:
It's application specific so can't be forced behind people's back. The `UNICODE` one is what people would get from from compiling UWP/WinRT with MSVC but not the other.